src/templating/templateEngine.js

```diff
@@ -37,11 +37,63 @@ ko.templateEngine.prototype['createJavaScriptEvaluatorBlock'] = function (script
 ko.templateEngine.prototype['makeTemplateSource'] = function(template, templateDocument) {
     // Named template
     if (typeof template == "string") {
+        var namedTemplateCache = this.namedTemplateCache || {};
+
+        // Cache entries last for 10s after the last access
+        var cacheLifetime = 10000;
+
+        // Registers a timer to clean up stale entries from the template cache.
+        function cleanupCache() {
+            // If we do not yet have a cleanup scheduled, schedule one
+            if (!this.templateCacheCleanupTimeoutID) {
```

Comment:
No, `this` there is `window`. `cleanupCache` function is called without context, so by default it's `window`. Also it looks like you can move this function out of `mateTemplateSource` method